packages/server/lib/util/socket_allowed.ts

```diff
@@ -36,7 +36,7 @@ export class SocketAllowed {
    */
   isRequestAllowed (req: Request) {
     const { remotePort, remoteAddress } = req.socket
-    const isAllowed = this.allowedLocalPorts.includes(remotePort!)
+    const isAllowed = (this.allowedLocalPorts.includes(remotePort!) || (!!process.env.CYPRESS_INTERNAL_E2E_TESTING_SELF && !!req.headers['x-cypress-forwarded-from-proxy']))
```

Comment:
can we pull this logic out into a function for clarity?

Reply:
Clarified the function a bit: https://github.com/cypress-io/cypress/commit/6ae7d07c26f712aaf612cf752d61cb1fecf5987f